crates/bitwarden-exporters/src/cxp/import.rs

```diff
@@ -0,0 +1,204 @@
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use chrono::{DateTime, Utc};
+use credential_exchange_types::format::{
+    Account as CxpAccount, BasicAuthCredential, Credential, Item, ItemType, PasskeyCredential,
+};
+
+use crate::{cxp::CxpError, CipherType, Fido2Credential, ImportingCipher, Login, LoginUri};
+
+pub(crate) fn parse_cxf(payload: String) -> Result<Vec<ImportingCipher>, CxpError> {
+    let account: CxpAccount = serde_json::from_str(&payload)?;
+
+    let items: Vec<ImportingCipher> = account.items.into_iter().flat_map(parse_item).collect();
+
+    Ok(items)
+}
+
+/// Convert a CXP timestamp to a DateTime<Utc>.
+///
+/// If the timestamp is None, the current time is used.
+fn convert_date(ts: Option<u64>) -> DateTime<Utc> {
+    ts.and_then(|ts| DateTime::from_timestamp(ts as i64, 0))
+        .unwrap_or(Utc::now())
+}
+
+fn parse_item(value: Item) -> Vec<ImportingCipher> {
+    let grouped = group_credentials_by_type(value.credentials);
+
+    let creation_date = convert_date(value.creation_at);
+    let revision_date = convert_date(value.modified_at);
+
+    match value.ty {
+        ItemType::Login => {
+            let basic_auth = grouped.basic_auth.first();
```

Comment:
We're only handling the first element in `Item`, but it looks like this function could handle more than one as returns a Vec, is this missing from the first version?

Reply:
Yes it's a known limitation of the current implementation.